### PR TITLE
Rewrite Windows 10 support

### LINKED ISSUE #487: Unable to connect on Windows
I am able to scan for bluetooth devices, but when I try to connect on Windows, there is a native popup saying "Connecting to <device>" followed by "Connection failed".

Stepping through the plugin JS, the error is logged from BluetoothLePlugin.js#L265, but the DevicePairingResultStatus code is 19: An unknown failure occurred.

My connection code is pretty simple:
```
app.bt_connect = function(address) {
	var params = { address: address };
	bluetoothle.connect(onConnect, onConnectError, params);
}
function onConnectError(result) {
	console.error(result);
}
```

### LINKED ISSUE #491: Windows: isEnabled not implemented
The `isEnabled` function isn't implemented.

I know that Windows functionality is experimental/limited, but it's a good idea to note the issues.

If I can, I may be able to contribute fixes to bring the Windows stuff up to snuff.

The command is:
`bluetoothle.isEnabled(function(result){console.log(result);});`

The error is:
> Missing Command Error

### LINKED ISSUE #501: Windows10 uwp
Hello,
Does the plugin work for windows10 uwp ?
Regards.
Fred

### LINKED ISSUE #508: Windows : unable to close
On windows 10 when I close the BLE connection, the device remains active.
I looked inside the plugin code and it looks like services are being closed but not the actual device.
Also the success callback is never called.

### LINKED ISSUE #509: Windows: why use ContainerId instead of MAC Address?
I'm working on getting our app that uses your great plugin to build on Windows 10 as an UWP project.

I'm finding that there's an issue with the way the "address" is handled. The plugin code treats the "ContainerId" as the device address instead of the MAC address.

This causes issues because of the way we have to get the DeviceInformation using `findAllAsync` which will wait for 30 seconds **every** time it tries to connect (because it's waiting for a timeout).

If we were to instead use the Device's MAC address, we could use `fromBluetoothAddressAsync` which is much quicker.

Are there design considerations that I'm not seeing for why ContainerId was chosen? If not, I'd like to rewrite the Windows 10 support to use MAC addresses instead. I think it'll be easier to maintain going forward.

### PR DESCRIPTION
**Pull request by TheBosZ**
This PR is to bring Windows 10 UWP support up to about the same level as Android and iPhone.

I did what I could to make the experience the same between the platforms.

We're using this plugin in an app that's targeting Android, iOS and Windows. With this PR, the app works as expected across the three.

The major missing functionality is advertisement data. I'm still working on that, but right now it appears as though the end-user has to jump through a lot of hoops to get it working so it may not be viable.

This closes #487, closes #491, closes #501, closes #508, and closes #509

### CODE REVIEW

File: src/windows/BluetoothLePlugin.js [hunk 1]
```diff
@@ -137,20 +145,14 @@ module.exports = {
...
-
     if (!WATCHER) {
       // watch BLE devices using device watcher.
-      var selector = Windows.Devices.Bluetooth.BluetoothLEDevice.getDeviceSelectorFromConnectionStatus(false);
-      WATCHER = DeviceInformation.createWatcher(selector, [NAME_KEY, RSSI_KEY, CONTAINER_ID_KEY]);
+      var selector = WindowsBluetooth.BluetoothLEDevice.getDeviceSelectorFromConnectionStatus(false);
+      if (params && params[0] && params[0].isConnectable) {
+        selector += ' AND System.Devices.Aep.Bluetooth.Le.IsConnectable:=System.StructuredQueryType.Boolean#True';
+      }
+      console.info('selector used is', selector);
```
Comment (randdusing) [inline src/windows/BluetoothLePlugin.js]: Can you remove this?

File: src/windows/BluetoothLePlugin.js [hunk 2]
```diff
@@ -225,62 +226,64 @@ module.exports = {
...
-    .then(function (devices) {
-      return Windows.Devices.Bluetooth.BluetoothLEDevice.fromIdAsync(devices[0].id);
-    })
-    .then(function (bleDevice) {
-      var DevicePairingProtectionLevel = Windows.Devices.Enumeration.DevicePairingProtectionLevel;
-      var DevicePairingResultStatus = Windows.Devices.Enumeration.DevicePairingResultStatus;
-
-      if (bleDevice.deviceInformation.pairing.isPaired) {
+    getDeviceByAddress(address)
+    // .then(function (bleDevice) {
```
Comment (randdusing) [inline src/windows/BluetoothLePlugin.js]: Did this get a lot simpler? Remove all the commented out / old code

Comment (TheBosZ) [reply src/windows/BluetoothLePlugin.js]: Yes, I was able to simplify it a ton. I'll remove the cruft.

File: src/windows/BluetoothLePlugin.js (hunk 1 above)
Comment (TheBosZ) [reply src/windows/BluetoothLePlugin.js]: I removed this and a bunch of other logging.

**Comment by randdusing** [comment]
New version 4.5.0 has been published to NPM

**Pull request merged**